scripts/code_style_rewrite_branch.py

```diff
@@ -0,0 +1,214 @@
+            return False
+
+    # Restyle the changed files to the new style
+    if code_style.fix_style(changed_files) != 0:
+        return False
+
+    # Add the changed files
+    for changed_file in changed_files:
+        result = subprocess.run(["git", "add", changed_file], check=False)
+        if result.returncode != 0:
+            print("Error adding changed file '" + changed_file + "'.", \
+                    file=sys.stderr)
+            return False
+
+    # Commit the newly added files
+    result = subprocess.run(["git", "commit", "--reuse-message=" + \
+            commit_hash], check=False)
+    if result.returncode != 0:
+        print("Error making restyled commit.", file=sys.stderr)
+        return False
+
+    return True
+
+def list_revisions(revision_or_range: str) -> List[str]:
+    """
+    Return the list of commits in revision_or_range.
+
+    If revision_or_range is a single revision, return it in a one-element
+    list. Otherwise return the list of commits in that range.
+
+    Return the list in order from oldest to newest.
+    """
+    result = subprocess.run(["git", "rev-list", "--no-walk", \
+            revision_or_range], stdout=subprocess.PIPE, check=False)
+    if result.returncode != 0:
+        print("Error getting revision list for '"+revision_or_range+"'.", \
+                file=sys.stderr)
+        return []
+
+    return list(reversed(str(result.stdout, "ascii").split()))
+
+def rewrite_branch_code_style(existing_branch: str, new_branch: str, \
+        codestyle_change_commit: str) -> bool:
+    """
+    Rewrite the given existing branch onto the given new branch on top of the
+    given commit that changes the code style.
+    """
+    # Create the new branch to rewrite onto at the code style change commit
+    result = subprocess.run(["git", "checkout", codestyle_change_commit], \
+            check=False)
+    if result.returncode != 0:
+        print("Error checking out code style commit '" + \
+                codestyle_change_commit + "'.", file=sys.stderr)
+        return False
+    result = subprocess.run(["git", "checkout", "-b", new_branch], check=False)
+    if result.returncode != 0:
+        print("Error creating destination branch '"+new_branch+"'.", \
+                file=sys.stderr)
+        return False
+
+    # List all the commits between the commit before the codestyle change
+    # (where the branch diverged) and the tip of the branch.
+    for rev in list_revisions(codestyle_change_commit+"~.."+existing_branch):
+        if not restyle_commit_onto_current(rev):
+            delete_branch(new_branch)
+            return False
+
+    return True
+
+def main() -> int:
+    """
+    Main with command line arguments.
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument('existing_branch', \
+            help='branch to rewrite with code style changes')
+    parser.add_argument('destination_branch', \
```

Comment:
Ultimately: the existing branch name will have to be reused for the restyled code, since you can change the branch name of a pull request.

Reply:
That's true, but I opted for a separate branch name in case something goes wrong during the rewrite - it seemed better to be nondestructive, and the user can then do:
```sh
git checkout original-branch
git reset --hard new-branch
```
As you say, it might be better to backup the old branch and overwrite the new one.

Reply:
This also points to the fact that we probably need some documentation to go with the script and explain how it fits in the general picture.

Reply:
Yes, I consider documentation part of the minimum delivery before we make the switch. I intend to write it this week.